src/db/pool.cr

```diff
@@ -62,6 +57,38 @@ module DB
       @initial_pool_size.times { build_resource }
     end
 
+    def reap_connection!(connection : T)
+      connection.health_check! if connection.responds_to?(:health_check!)
```

Comment:
this needs to remove the connection from `@idle` otherwise we've got some possibility of race conditions, I think